PYME/Acquire/SpoolController.py

```diff
@@ -482,13 +522,43 @@ def stop_spooling(self):
         return 'OK'
 
     @webframework.register_endpoint('/start_spooling', output_is_json=False)
-    def start_spooling(self, filename=None, max_frames=sys.maxsize):
-        self.spool_controller.StartSpooling(fn=filename, maxFrames=max_frames)
+    def start_spooling(self, body='{}'):
```

Comment:
this completely changes the signature. How does this effect the web front end? Is there a strong argument for getting rid of the query arguments? Using query arguments makes it much easier to debug and/or write an HTML frontend for.

Reply:
I tested it with the web front end; as far as I could tell there's no change.

I googled and it sounded like once you're passing a dictionary via query you might as well pass a dictionary. It also makes the wrapper pass-through, which I think is nice. I did not have HTML frontend in mind, nor do I have experience with it, as you know

Reply:
In terms of web front-end, putting things in a json encoded body means that you can only call it via javascript (and will need to do some javascript munging to get the parameters into json). If you keep things with query arguments, you can easily make it the target of a link or button without doing anything in js. You can also test it out by simply pointing your web-browser to the endpoint with the relevant arguments in the URL. Past a certain complexity it makes sense to format things as json, but I'm not sure that applies for most start_spooling() calls.

Reply:
Opening a python console and importing requests is pretty minimal overhead for testing. json standing for JavaScript Object Notation I would have assumed it would be pretty easy to work with in javascript.

In practice, I'm not going to lose sleep over this and if you prefer query that's OK by me, but my opinion is that passing all 8 of our kwargs (since we don't know the state of the SpoolController when our tasks becomes active) is not very clean.

Reply:
I think you're right that most calls will be much less than all kwargs in StartSpooling, but if one doesn't know the state of the spoolcontroller they might fill out quite a few each time. If that's OK then I'll revert it to kwargs. Do you want me to add the rest of the StartSpooling kwargs (pep8 on the wrapper to current kwargs on the `self.spool_controller.StartSpooling` call) while I'm at it?